Client/components/signup/SignupForm.js

```diff
@@ -32,6 +34,25 @@ class SignupForm extends React.Component {
     }
     return isValid;
   }
+  checkUserExists(e) {
+    const field = e.target.name;
+    const val = e.target.value;
+    if (val !== '') {
+      this.props.isUserExists(val).then(res => {
+        let errors = this.state.errors;
+        let invalid;
+        if (res.data.user) {
+          errors[field] = 'There is user with such ' + field;
+          invalid = true;
+        } else {
+          errors[field] = '';
+          invalid = false;
+        }
+        this.setState({ errors, invalid });
```

Comment:
this is okay for presentation purposes, but son't forget to use redux.
this kind of state should be inside of a redux store (so you can see it in the redux logger).
There needs to be a container that injects the state into props of components like this one.
And actions which are called in components like this one.
`CheckIfUserExists` would be a good candidate for a redux action.
it would add this error (if there is one) into a reducer and then the container would pick it up and display it inside of any component that needs to know about this error.